File: system-tests/fixtures/api/complex-api/IOS/SystemTestEventAPI.swift
#if swift(>=4.0)
@objcMembers public class SystemTestEventAPI: NSObject {
    static let kEventTestEvent = "com.complex.ern.api.event.testEvent"

    public lazy var events: SystemTestEventAPIEvents = {
        SystemTestEventEvents()
    }()

    public lazy var requests: SystemTestEventAPIRequests = {
        SystemTestEventRequests()
    }()
}

@objcMembers public class SystemTestEventAPIEvents: NSObject {
    public func addTestEventEventListener(eventListener _: @escaping ElectrodeBridgeEventListener) -> UUID? {
        assertionFailure("should override")
        return UUID()
    }

    public func removeTestEventEventListener(uuid _: UUID) -> ElectrodeBridgeEventListener? {
        assertionFailure("should override")
        return nil
    }

    public func emitEventTestEvent(buttonId _: String) {
        assertionFailure("should override")
    }
}

@objcMembers public class SystemTestEventAPIRequests: NSObject {
}

#else

public class SystemTestEventAPI: NSObject {
    static let kEventTestEvent = "com.complex.ern.api.event.testEvent"

    public lazy var events: SystemTestEventAPIEvents = {
        SystemTestEventEvents()
    }()

    public lazy var requests: SystemTestEventAPIRequests = {
        SystemTestEventRequests()
    }()
}

public class SystemTestEventAPIEvents: NSObject {
    public func addTestEventEventListener(eventListener _: @escaping ElectrodeBridgeEventListener) -> UUID? {
        assertionFailure("should override")
        return UUID()
    }

    public func removeTestEventEventListener(uuid _: UUID) -> ElectrodeBridgeEventListener? {
        assertionFailure("should override")
        return nil
    }

    public func emitEventTestEvent(buttonId _: String) {
        assertionFailure("should override")
    }
}

public class SystemTestEventAPIRequests: NSObject {
}
#endif

File: system-tests/fixtures/api/complex-api/android/lib/src/main/java/com/complex/ern/api/SystemTestEventApi.java
/*
 * Copyright 2020 Walmart Labs
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.complex.ern.api;

import androidx.annotation.NonNull;

import com.walmartlabs.electrode.reactnative.bridge.ElectrodeBridgeEvent;
import com.walmartlabs.electrode.reactnative.bridge.ElectrodeBridgeEventListener;
import com.walmartlabs.electrode.reactnative.bridge.ElectrodeBridgeRequestHandler;
import com.walmartlabs.electrode.reactnative.bridge.ElectrodeBridgeResponseListener;
import com.walmartlabs.electrode.reactnative.bridge.None;
import com.walmartlabs.electrode.reactnative.bridge.RequestHandlerHandle;

import java.util.*;
import java.util.UUID;


public final class SystemTestEventApi {
    private static final Requests REQUESTS;
    private static final Events EVENTS;

    static {
        REQUESTS = new SystemTestEventRequests();
        EVENTS = new SystemTestEventEvents();
    }

    private SystemTestEventApi() {
    }

    @NonNull
    public static Requests requests() {
        return REQUESTS;
    }

    @NonNull
    public static Events events() {
        return EVENTS;
    }

    public interface Events {
        String EVENT_TEST_EVENT = "com.complex.ern.api.event.testEvent";

        UUID addTestEventEventListener(
                @NonNull final ElectrodeBridgeEventListener<String> eventListener);

        ElectrodeBridgeEventListener<ElectrodeBridgeEvent> removeTestEventEventListener(
                @NonNull final UUID uuid);

        void emitTestEvent(@NonNull String buttonId);
    }

    public interface Requests {
    }
}
